bigframes/core/compile/compiled.py

```diff
@@ -604,6 +610,60 @@ def reversed(self) -> OrderedIR:
         expr_builder.ordering = self._ordering.with_reverse()
         return expr_builder.build()
 
+    def aggregate(
```

Comment:
@chelsea-lin Can you help me understand more about where this came from? Presumably we already support aggregates, right, so I'm wondering:

1. what this does that the existing code paths can't handle and
2. whether those existing code paths are now redundant and should be removed / migrated to this

Reply:
The `array_agg` is the first aggregated method, which requires `OrderedIR` so that a new `aggregate()` method is added to `OrderedIR`. Good catch! I just remove the redundant codes in the new commit.